src/bvals/cc/bvals_cc.cpp

```diff
@@ -159,7 +161,8 @@ int CellCenteredBoundaryVariable::LoadBoundaryBufferToCoarser(Real *buf,
   int p = 0;
   pmb->pmr->RestrictCellCenteredValues(var_cc, coarse_buf, nl_, nu_, si, ei, sj, ej, sk,
                                        ek);
-  BufferUtility::PackData(coarse_buf, buf, nl_, nu_, si, ei, sj, ej, sk, ek, p);
+  ParArray4D<Real> coarse_buf_ = coarse_buf.Get<4>(); // auto template deduction fails
```

Comment:
Do we really need to pull out the ParArray4D?  Can't we just pass the ParArrayND?  In fact, we may *want* something more than 4D.  What if the variable is a phase space density, for example.

Reply:
Obviously, the PackData function expects something 4D, so I get why you did this.  My question is really is this necessary?  I suppose we'll want to move all this over to using VariablePacks (or something like it) to combine this stuff into fewer/bigger kernels, so maybe it's not worth thinking to hard at the moment.

Reply:
This is related to our discussion during the last call.
I prefer to use strong types in the interfaces (also over assert statements) where the implementation expects a a certain type and would be happy to refactor again if the use case changes.
My hope is that this prevents future bugs as the compiler will shout and the functions are harder to use in an unintended way.

Reply:
Presumably at some point soon, we will need to take a second look at this code and use the variable packs that we developed?

Reply:
yes, depending on what the profiling will tell us I can also see that these function may look (completely) different when we start optimizing the code.